File: cs50/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string words);
int count_sentences(char *sent);

int main(void)
{

    string all = get_string("Text: ");

    int letters = count_letters(all);
    int words = count_words(all);
    int sent = count_sentences(all);

    float L = ((letters/(float)words) * 100);
    float S = ((sent/(float)words) * 100);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
        {
            printf("Grade 16+\n");
        }
    else if (index < 1)
        {
            printf("Before Grade 1...");
        }
    else
        {
            printf("Grade %i\n", index);
        }
}

int count_letters(string text)
{
    int num = strlen(text);
    int len = 0;

    for (int i = 0; i < num; i++)
    {
        if (isalpha(text[i]))
        {
            len ++;
        }
    }

return len;
}

int count_words(string word)
{
    int len = 1;
    int num = strlen(word);

    for (int i = 0; i < num; i++)
    {
        if (isspace(word[i]))
        {
            len ++;
        }
    }
return len;
}

int count_sentences(char *sent)
{
    int len = 0;
    int num = strlen(sent);

    for (int i = 0; i < num; i++)
        {
            if ((sent[i]) == '.' || (sent[i]) == '!' || (sent[i]) == '?')
            {
                len ++;
            }
        }
return len;
}

File: cs50/sentimental-readability/readability.py
from cs50 import get_string
import re
import nltk

def main():

    all = get_string("Text: ")

    index = 0.0588 * L - 0.296 * S - 15.8



    letters = count_letters(all)
    words = count_words(all)
    sent = count_sentences(all)

def count_letters(text):

    letters = len(text)
    return letters

def count_words(text):

   word_re = re.compile(r'\b\w+\b')
   word_count = len(word_re.findall(text))
   return word_count

def count_sentences(text):

    sentences= nltk.sent_tokenize(text)
    length= len(sentences)




main()